examples/simple/igraph_get_all_shortest_paths_dijkstra.c

```diff
@@ -165,21 +188,30 @@ int main() {
     igraph_vector_view(&weights_vec, weights2, sizeof(weights2) / sizeof(igraph_real_t));
     igraph_get_all_shortest_paths_dijkstra(
                 &g,
-                /*res=*/ &res, /*nrgeo=*/ &nrgeo,
+                /*vertices=*/ &vertices, /*edges=*/ &edges, /*nrgeo=*/ &nrgeo,
                 /*from=*/ 0, /*to=*/ vs,
                 /*weights=*/ &weights_vec, /*mode=*/ IGRAPH_OUT);
 
-    check_nrgeo(&g, vs, &res, &nrgeo);
+    check_nrgeo(&g, vs, &vertices, &nrgeo);
 
     /* Sort the paths in a deterministic manner to avoid problems with
      * different qsort() implementations on different platforms */
-    igraph_vector_ptr_sort(&res, vector_tail_cmp);
+    igraph_vector_ptr_sort(&vertices, vector_tail_cmp);
+
+    for (i = 0; i < igraph_vector_ptr_size(&vertices); i++) {
+        igraph_vector_print(VECTOR(vertices)[i]);
+        igraph_vector_destroy(VECTOR(vertices)[i]);
+        igraph_free(VECTOR(vertices)[i]);
+        VECTOR(vertices)[i] = 0;
+    }
+
+    igraph_vector_ptr_sort(&edges, vector_tail_cmp);
```

Comment:
This is not correct.

The idea was to return the paths sorted by target vertex, i.e. effectively grouped by target vertex. This code sorts the edge result _differently_ than the vertex result, which must not be the case. The `i`th element of `vertices` must correspond to the same path as the `i`th element of `edges`. Sorting by the final edge ID doesn't really make sense.

We must be careful not to blindly copy code, but understand why it was so in the first place.

See also:

https://igraph.discourse.group/t/why-are-shortest-paths-sorted/626

Reply:
I was wondering about this too -- I was simply looking at the diff of the expected output of the test case and I did not understand why the order of the edgeID-paths did not match the vertexID-paths.

On the other hand, aren't we sorting the paths based on the ID of the last vertex anyway (in `igraph_get_all_shortest_paths_dijkstra()`)? If we do, we don't need to sort again in the example.

Reply:
Okay, now I understand. The implementation in `igraph_get_all_shortest_paths_dijkstra()` does not do a full lexicographic ordering based on the vertex IDs from the end, it simply compares the trailing vertex IDs, but if there is a tie, it is free to order the paths in any arbitrary order. The extra sorting in the test case ensures that ties are also broken in a deterministic manner.

I will probably move this example to the `tests/unit` folder if we have merged this PR as it is overly detailed and complex to serve as a good example. We'll need to write a shorter example for the manual.

Reply:
> The ith element of vertices must correspond to the same path as the ith element of edges. Sorting by the final edge ID doesn't really make sense.

I think that for the sake of the test case and ensuring determinism in reporting the output it's okay.

On the other hand, we should actually test whether the ordering of the vectors in `vertices` and `edges` is consistent, i.e. if we ask for both, igraph must return them in an order that resolving edge IDs in the i-th vector of `edges` yields the same vertices in the same order as they are placed in the i-th vector of `vertices`. This must be tested explicitly instead of simply printing the output of `edges` to stdout because the extra post-processing (sorting) in the test case would hide problems where the ordering of the vectors in `vertices` and `edges` is not consistent.

Reply:
Hi @szhorvat,
we actually don't need to sort the edge result (and the vertices result) because the function `igraph_get_all_shortest_paths_dijkstra` return a sorted edge result vector (based on the target node and sorted the same as the vertex retult vector).
@ntamas should I move the example to `test/unit` and write a short example?
thanks